core/src/main/java/org/kohsuke/stapler/lang/Klass.java

```diff
@@ -56,14 +57,15 @@ public Class toJavaClass() {
         return navigator.getDeclaredMethods(clazz);
     }
 
+    /**
+     * Gets list of fields declared by the class.
+     * @return List of fields. 
+     *         May return empty list in the case of obsolete {@link #navigator}, which does not offer the method.
+     * @since 1.246
+     */
+    @Nonnull
     public List<FieldRef> getDeclaredFields() {
-        try {
-            return navigator.getDeclaredFields(clazz);
-        } catch (AbstractMethodError err) {
-            // A plugin uses obsolete version of Stapler-dependent library (e.g. JRuby), which does not offer the method (JENKINS-39414)
-            // TODO: what to do with Logging? The error must be VERY visible, but it will totally pollute system logs
-            return Collections.emptyList();
-        }
+        return navigator.getDeclaredFields(clazz);
```

Comment:
It was a hack in #85 in order to suppress the binary compatibility issue. Now we solve it by providing default implementations in `KlassNavigator`. Hence this `AbstractMethodError` handler is no longer required